### PR TITLE
Centralize on a single garbage collection helper

### LINKED ISSUE #36926: Centralize garbage collection
Patch https://github.com/web-platform-tests/wpt/commit/b181b1cf64799f29527a4a0a1d5a2bfadb69c5d1 made a few improvements to how tests do garbage collection.  

@saschanaz suggested we centralize that somewhere... 

@domenic wrote 
Approving since this is a strict improvement, but some suggestions:

- Copy the more informative comments from streams/
- Use the self.garbageCollect = () => { } format since it makes it clearer that it's adding a global.
- Consolidate these all into /common/gc.js
- Fix the [others](https://github.com/search?q=repo%3Aweb-platform-tests%2Fwpt%20%22function%20garbageCollect%22&type=code) found in permissions/ and fs/, and [the ones](https://github.com/search?q=repo%3Aweb-platform-tests%2Fwpt+%22function+runGarbageCollection%22&type=code) found in webusb/ and bluetooth/, and finally [the one found in js/builtins](https://github.com/web-platform-tests/wpt/blob/0802d8bd01e17178d9cd2355e6fbaf3c098e7953/js/builtins/weakrefs/resources/maybe-garbage-collect.js#L13).
- Fix [various cases that aren't using this, to use this](https://github.com/search?q=repo%3Aweb-platform-tests%2Fwpt+%22https%3A%2F%2Fgithub.com%2Fweb-platform-tests%2Fwpt%2Fissues%2F7899%29%22&type=code)

### PR DESCRIPTION
**Pull request by domenic**
Closes #36926.

---

Unfortunately lots of changes to the bluetooth/ directory due to the generate.py script not being run in a while...

### CODE REVIEW

**Review by miketaylr — APPROVED** [review]
LGTM

(I was initially confused why this adds `/common/gc.js` to so many bluetooth tests that seemingly don't call `garbageCollect` - but I see now that `resources/bluetooth-test.js` exposed a `runGarbageCollection` method that might be used in other tests.)

**Comment by inexorabletash** [comment]
Should there be docs in `common/gc.js` and/or on https://web-platform-tests.org/ documenting how to use and how _not_ to use `garbageCollect()` ? e.g. since it's "best effort" you shouldn't rely on collection being successful.

(e.g. for those rare specs where GC ends up being semi-observable because a resource is released, this doesn't let you write tests that ensure that behavior)

**Review by marcoscaceres — APPROVED** [review]
Thanks for doing this @domenic. I'm hopeful that garbageCollect() can then go away entirely once we have `TestUtils.gc()` everywhere.

File: js/builtins/weakrefs/resources/maybe-garbage-collect.js [hunk 1]
```diff
@@ -2,33 +2,8 @@
...
-    await GCController.collect();
-  } else {
-    /* eslint-disable no-console */
-    console.warn('Tests are running without the ability to do manual ' +
-                 'garbage collection. They will still work, but ' +
-                 'coverage will be suboptimal.');
-    /* eslint-enable no-console */
-  }
-}
+self.maybeGarbageCollectAsync = garbageCollect;
```
Comment (Ms2ger) [inline js/builtins/weakrefs/resources/maybe-garbage-collect.js:6]: Any reason to keep this alias?

File: common/gc.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * environment-specific nonstandard functions, with a final result of just
+ * creating a lot of garbage (in which case you will get a console warning).
+ *
+ * This should generally only be used to attempt to trigger bugs and crashes
+ * inside tests, i.e. cases where if garbage collection happened, then this
+ * should not trigger some misbehavior. You cannot rely on garbage collection
+ * successfully trigger, or that any particular unreachable object will be
+ * collected.
+ *
+ * @returns {Promise<undefined>} A promise you should await to ensure garbage
```
Comment (Ms2ger) [inline common/gc.js:13]: It doesn't seem like this can resolve asynchronously. Is the return type future-proofing or is there a path that I missed?

**Review by Ms2ger — APPROVED** [review]
Thanks. I would have preferred that you didn't reformat a bunch of tests, though.

File: common/gc.js (hunk 2 above)
Comment (domenic) [reply common/gc.js:13]: All the `return x()` cases are calling promise-returning functions.

File: js/builtins/weakrefs/resources/maybe-garbage-collect.js (hunk 1 above)
Comment (domenic) [reply js/builtins/weakrefs/resources/maybe-garbage-collect.js:6]: It seemed a bit weird to break the naming scheme that the other helpers in this file use, so I thought it'd be least disruptive to keep the alias.

**Comment by domenic** [comment]
> Thanks. I would have preferred that you didn't reformat a bunch of tests, though.

Me too, but the generate.py script had ideas of its own.

**Pull request merged**